eng/vmr-sync.sh

```diff
@@ -1,63 +1,47 @@
-### the 'VMR Source-Build'.
-### The following scenario is assumed:
-### - There is a PR in dotnet/installer
-### - The PR is failing on the 'VMR Source-Build' job in the 'Synchronize dotnet/dotnet' step
+### This script is used for synchronizing the dotnet/dotnet VMR locally. This means pulling new
+### code from various repositories into the 'dotnet/dotnet' repository.
 ###
-### There are a few possible reasons but usually this happens because there is a mismatch between
-### source patches that are applied on top of the files that are being synchronized from repositories
-### into the 'dotnet/dotnet' repo and new changes in the repositories.
-### This manifests as the following error:
-###     fail: Failed to synchronize repo installer
-###           Failed to apply the patch for src/aspnetcore
-###           Exit code: 1
-###           Std err:
-###           error: patch failed: src/aspnetcore/eng/SourceBuild.props:55
-###           error: src/aspnetcore/eng/SourceBuild.props: patch does not apply
+### The script is used during CI to ingest new code based on dotnet/installer but it can also help
+### for reproducing potential failures during installer's PRs, namely to fix the Source-Build.
+### Another usecase is to try manually synchronizing a given commit of some repo into the VMR and
+### trying to Source-Build the VMR. This can help when fixing the Source-Build but using a commit
+### from a not-yet merged branch (or fork) to test the fix will help.
 ###
-### where 'src/aspnetcore/eng/SourceBuild.props' would be the file that is being patched and new
-### changes to it are conflicting with the patch that is being applied.
-###
-### The whole process can be reproduced locally easily by running this script.
-### The patches are located in a folder in the 'dotnet/installer' repository.
-### At the moment of writing, the location is 'src/SourceBuild/patches' but to get
-### the up-to-date location, please see the 'patchesPath' property in
-### https://github.com/dotnet/dotnet/blob/main/src/source-mappings.json
-###
-### You will need to compare what is in the patch and what is in the file and fix the patch.
-###
-### The tooling that synchronizes the VMR will need to clone the various repositories.
-### It clones them into a temporary folder and re-uses them on future runs so it is advised
-### you dedicate a folder to this to speed up your re-runs.
-###
-### This script will synchronize the 'dotnet/dotnet' repo locally and let you inspect the changes.
+### The tooling that synchronizes the VMR will need to clone the various repositories into a temporary
+### folder. These clones can be re-used in future synchronizations so it is advised you dedicate a
+### folder to this to speed up your re-runs.
 ###
 ### USAGE:
-###     ./vmr-sync.sh --tmp-dir "$HOME/repos/tmp"
+###   ./vmr-sync.sh --repository runtime:e7e71da303af8dc97df99b098f21f526398c3943 --tmp-dir "$HOME/repos/tmp"
 ### Options:
+###   --repository name:GIT_REF
```

Comment:
Making this required feels like a UX regression for the canonical dev workflow of wanting to sync the entire VMR from installer.  Can't this be optional and default to the installer repo?

Reply:
I can totally bring that default back.

My thinking was that with the codespace syncing the PR for you, there would be no scenario where you'd need to sync installer recursively yourself. But I understand I can be wrong and if you do use this locally then it makes sense to have this default.

Reply:
Please bring it back.  There are several workflows I perform outside of codespace.